### PR TITLE
Replacing formatters

### PR DESCRIPTION
**Pull request by AndreasArvidsson**
By request I'm updating the formatters in community with my own implementation. 

* Each formatter can decide how it wants to unformat the text beforehand when doing reformat
* Allow you to reformat strings and still keep delimiters
* Added proper reformat action

All the issues around updating the history in the capture in having actions like `insert_many` I leave to a follow up. We all show probably should have a follow up around the spoken forms and where they are located. Better outlined here: https://github.com/talonhub/community/issues/1418

### CODE REVIEW

**Comment by AndreasArvidsson** [comment]
Waiting for talonfmt to be updated for latest python version
https://github.com/talonhub/community/pull/1430

**Comment by nriley** [comment]
We discussed this in the backlog session.

Things look good right now but for a few new pieces we will need to add for the first version of this PR:

* Since we now have differences between formatting and reformatting, it will be helpful to have "format help" allow you to specify a different starting string other than the existing "one two three". Format help should also show the input string in addition to the formatted string.
* We need to pick some spoken forms for the new "reformatter" formatters.

Things that are explicitly out of scope for this version of the PR:

* Allowing extensibility of formatters from outside of formatters.py
* Using a Cursorless-like CSV to allow customization of spoken forms, including the ability to compose formatters with a single spoken form

File: core/text/formatters.py [hunk 1]
```diff
@@ -181,29 +308,20 @@ def every_word(word_func):
     "kebab": "DASH_SEPARATED",
     "packed": "DOUBLE_COLON_SEPARATED",
     "padded": "SPACE_SURROUNDED_STRING",
-    "slasher": "ALL_SLASHES",
-    "conga": "SLASH_SEPARATED",
+    "lasher": "SLASH_SEPARATED",
```
Comment (nriley) [inline core/text/formatters.py:311]: These are actually different formatters.

**Review by nriley — CHANGES_REQUESTED** [review]

File: core/text/formatters.py [hunk 2]
```diff
@@ -181,29 +308,20 @@ def every_word(word_func):
...
     "smash": "NO_SPACES",
     "snake": "SNAKE_CASE",
     "string": "SINGLE_QUOTED_STRING",
 }
+
 prose_formatter_names = {
     "say": "NOOP",
     "speak": "NOOP",
-    "sentence": "CAPITALIZE_FIRST_WORD",
+    "sentence": "SENTENCE",
```
Comment (nriley) [inline core/text/formatters.py:320]: `CAPITALIZE_FIRST_WORD` is in use in several other places and should be preserved even if it is a misnomer. (Sorry!)

File: core/text/formatters.py (hunk 1 above)
Comment (AndreasArvidsson) [reply core/text/formatters.py]: done

File: core/text/formatters.py (hunk 2 above)
Comment (AndreasArvidsson) [reply core/text/formatters.py]: done

**Comment by AndreasArvidsson** [comment]
>     * We need to pick some spoken forms for the new "reformatter" formatters.
Done

>     * Since we now have differences between formatting and reformatting, it will be helpful to have "format help" allow you to specify a different starting string other than the existing "one two three". Format help should also show the input string in addition to the formatted string.
I'm worried it will be confusing if we try to show a new format and a reformat in the same line. I'm wondering if this is something we should have a discussion about in a future meet up and then do in a separate pull request?

**Comment by nriley** [comment]
> > ```
> > * Since we now have differences between formatting and reformatting, it will be helpful to have "format help" allow you to specify a different starting string other than the existing "one two three". Format help should also show the input string in addition to the formatted string.
> > ```
> 
> I'm worried it will be confusing if we try to show a new format and a reformat in the same line. I'm wondering if this is something we should have a discussion about in a future meet up and then do in a separate pull request?

Have you looked at how it looks in the current version? I can't even get `format help` to display with your version:

```
 in script at /Users/nicholas/.talon/user/talon_community/core/help/help.talon:10:
 > user.help_formatters(user.get_formatters_words())
KeyError: 'ALL_UPPERCASE'
```

**Comment by AndreasArvidsson** [comment]
> > > ```
> > > * Since we now have differences between formatting and reformatting, it will be helpful to have "format help" allow you to specify a different starting string other than the existing "one two three". Format help should also show the input string in addition to the formatted string.
> > > ```
> > 
> > 
> > I'm worried it will be confusing if we try to show a new format and a reformat in the same line. I'm wondering if this is something we should have a discussion about in a future meet up and then do in a separate pull request?
> 
> Have you looked at how it looks in the current version? I can't even get `format help` to display with your version:
> 
> ```
>  in script at /Users/nicholas/.talon/user/talon_community/core/help/help.talon:10:
>  > user.help_formatters(user.get_formatters_words())
> KeyError: 'ALL_UPPERCASE'
> ```

I forgot that in community it's all lower case, but all caps. Fixed now.

**Comment by AndreasArvidsson** [comment]
> Thanks. I'm out of time for Talon work this weekend but can take a look later; others can of course feel free to offer suggestions! My concern is that if we have multiple formatters that appear to do the same thing using our existing help system that it'll be even more confusing for new users than it is right now with the existence of prose and code formatters.

I agree, but I'm definitely not sure how to do this in a non confusing way with the current help system.

**Comment by nriley** [comment]
OK. I've got some ideas so can work on prototyping them when I have time (may not be for a couple weeks as I will be working next weekend).

**Comment by AndreasArvidsson** [comment]
We can just disabled the new capitalize formatter and re-enable that when we have a better help menu?

**Comment by AndreasArvidsson** [comment]
Note from last community session. Split formatter help into format and reformat.

**Comment by AndreasArvidsson** [comment]
@nriley I have now added a separate help for reformatting.

**Comment by AndreasArvidsson** [comment]
@nriley How have your testing turned out?

**Comment by nriley** [comment]
Very good - I've been using it for weeks now! Sorry, thought we had already merged this one...

**Review by nriley — APPROVED** [review]

**Pull request merged**